### PR TITLE
refactor: wrapper should not have internal state

### PR DESCRIPTION
**Pull request by HiroIshida**
I think DetecWrapper should be just a function, and should not have any internal state. Based on this concept, the source code is refactored. 

@Affonso-Gui

### CODE REVIEW

File: node_script/wrapper.py [hunk 1]
```diff
@@ -17,13 +18,62 @@
...
+        debug_img = _cv_bridge.cv2_to_imgmsg(
+            self.visualization.get_image(), encoding="rgb8")
+        debug_img.header = self.header
+        return debug_img
+
+    def get_ros_debug_segmentation_img(self) -> Image:
+        human_friendly_scaling = 255 // self.segmentation_raw_image.max()
+        new_data = (self.segmentation_raw_image * human_friendly_scaling).astype(np.uint8)
+        debug_seg_img = _cv_bridge.cv2_to_imgmsg(new_data, encoding="mono8")
+        assert self.header is not None
```
Comment (Affonso-Gui) [inline node_script/wrapper.py:50]: Should we add this assertion on other methods as well?
Or does the typing `header: Header` (as opposed to `Optional[Header]`) already ensure that the header is never None?

File: node_script/wrapper.py [hunk 2]
```diff
@@ -83,68 +129,15 @@ def inference_step(self, msg: Image) -> Tuple[Image, List[int], List[float], Opt
...
-            debug_seg_img = None
-
-        return seg_info, debug_img, debug_seg_img
+        result = InferenceRawResult(
+            data,
+            class_indexes,
+            scores,
+            visualized_output,
+            msg.header,
+            self.class_names)
```
Comment (Affonso-Gui) [inline node_script/wrapper.py:142]: Should we pass directly `[self.class_names[i] for i in class_indexes]`? 
Or do we really want to keep the full vocabulary?

**Comment by Affonso-Gui** [comment]
Cool! Having a separate class for the results makes it much cleaner. Thanks for taking the time to fix this!

File: node_script/wrapper.py (hunk 1 above)
Comment (HiroIshida) [reply node_script/wrapper.py:50]: True. header is ensured to be not None, so we don't need this assertion
I will fix

File: node_script/wrapper.py (hunk 2 above)
Comment (HiroIshida) [reply node_script/wrapper.py:142]: That's true. I will fix this

**Comment by HiroIshida** [comment]
@Affonso-Gui 
Thanks for you review
I reflect your review in https://github.com/HiroIshida/detic_ros/commit/e3596bd88c02f9f99f4dac2688741b6d886e2b57

**Pull request merged**